flink-runtime/src/main/java/org/apache/flink/runtime/asyncprocessing/declare/state/StateWithDeclaredNamespace.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.asyncprocessing.declare.state;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.api.common.state.v2.State;
+import org.apache.flink.runtime.asyncprocessing.declare.DeclaredVariable;
+import org.apache.flink.runtime.state.v2.internal.InternalAggregatingState;
+import org.apache.flink.runtime.state.v2.internal.InternalListState;
+import org.apache.flink.runtime.state.v2.internal.InternalMapState;
+import org.apache.flink.runtime.state.v2.internal.InternalPartitionedState;
+import org.apache.flink.runtime.state.v2.internal.InternalReducingState;
+import org.apache.flink.runtime.state.v2.internal.InternalValueState;
+
+/**
+ * State wrapped by declared namespace. This is useful for window operator, where we need to
+ * frequently setup current namespace in each lambda. This wrap class help use to invoke {@code
```

Comment:
Suggestion:

A partitioned state that wraps a declared namespace and hide the namespace switching from user. User will use the state just like the public state APIs without any namespaces manipulation. Useful in window operators.

Reply:
Resolved. 

Changed to:
```
/**
 * A partitioned state that wraps a declared namespace and hide the namespace switching from user. 
 * User will only use the state just like the public state APIs without any consideration on namespace. 
 * 
 * <p>This wrap class is useful in DataStream window operation, where namespace is managed by the operator
 * and user function is free from namespace manipulation.
 */
```